runtime/drivers/duckdb/model_manager.go

```diff
@@ -0,0 +1,176 @@
+package duckdb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+type ModelInputProperties struct {
+	SQL string `mapstructure:"sql"`
+}
+
+func (p *ModelInputProperties) Validate() error {
+	if p.SQL == "" {
+		return fmt.Errorf("missing property 'sql'")
+	}
+	return nil
+}
+
+type ModelOutputProperties struct {
+	Table               string                      `mapstructure:"table"`
+	Materialize         *bool                       `mapstructure:"materialize"`
+	UniqueKey           []string                    `mapstructure:"unique_key"`
+	IncrementalStrategy drivers.IncrementalStrategy `mapstructure:"incremental_strategy"`
+}
+
+func (p *ModelOutputProperties) Validate(opts *drivers.ModelExecutorOptions) error {
+	if opts.Incremental {
+		if p.Materialize != nil && !*p.Materialize {
+			return fmt.Errorf("incremental models must be materialized")
+		}
+		p.Materialize = boolPtr(true)
+	}
+
+	if opts.InputConnector != opts.OutputConnector {
+		if p.Materialize != nil && !*p.Materialize {
+			return fmt.Errorf("models that output to a different connector must be materialized")
+		}
+		p.Materialize = boolPtr(true)
+	}
+
+	switch p.IncrementalStrategy {
+	case drivers.IncrementalStrategyUnspecified, drivers.IncrementalStrategyAppend, drivers.IncrementalStrategyMerge:
+	default:
+		return fmt.Errorf("invalid incremental strategy %q", p.IncrementalStrategy)
+	}
+
+	if p.IncrementalStrategy == drivers.IncrementalStrategyMerge && len(p.UniqueKey) == 0 {
+		return fmt.Errorf(`must specify a "unique_key" when "incremental_strategy" is %q`, p.IncrementalStrategy)
```

Comment:
**Issue**: The error message when the 'unique_key' is missing for a 'merge' strategy is not formatted correctly for clarity. <br> **Fix**: Improve the clarity of the error message by adjusting the wording and format. <br> **Code Suggestion**: 
 ```
 - return fmt.Errorf('must specify a "unique_key" when "incremental_strategy" is %q', p.IncrementalStrategy)
 + return fmt.Errorf("A 'unique_key' must be specified when the 'incremental_strategy' is set to 'Merge'.")
 ```